insert_libzstd_ifdefs.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
```

Comment:
Do you mind putting this file in a `tools/` folder ?
That way it's much easier to discover (and we can add later on a script to pull the latest zstd c upstream)